lib/screens/drawing_screen.dart

```diff
@@ -33,6 +38,9 @@ class DrawingScreenState extends State<DrawingScreen> {
 
   void _initData() async {
     _activity = await ActivityRepo().getActivity(widget.activityId);
+    User _user = AppStateContainer.of(context).state.loggedInUser;
+    print(await ActivityProgressRepo()
+        .insertActivityProgress(_user.id, widget.topicId, widget.activityId));
```

Comment:
What will happen if the user attempts the same activity multiple times?

Reply:
Nothing just it will check if not there then insert.